whelktool/scripts/cleanups/2021/06/lxl-3640-add-mimer-uri-identifiers/script.groovy

```diff
@@ -0,0 +1,57 @@
+List idList = new File(scriptDir, 'idlist.yaml').readLines()
+
+Map mimerUriById = [:]
+
+String id
+
+idList.each { String line ->
+    if (line =~ "libris-id:")
+        id = line.replaceFirst(/.*libris-id:/, "").trim()
+    else if (line =~ /uri:/) {
+        String uri = line.replace("uri:", "").trim()
+
+        if (id.isNumber()) {
+            String where = """
+                collection = 'bib' 
+                AND deleted = 'false' 
+                AND data#>>'{@graph,0,controlNumber}' = '${id}'
+            """
+
+            selectBySqlWhere(where) {
+                id = it.doc.shortId
+            }
+        }
+
+        mimerUriById[id] = uri
+    }
+}
+
+selectByIds(mimerUriById.keySet()) { data ->
+    Map instance = data.graph[1]
+    List identifiedBy = instance.identifiedBy
+
+    Map identifier =
+            [
+                "@type": "Identifier",
+                "typeNote": "uri",
```

Comment:
Is this `typeNote` to get desired MARC out of this? Otherwise I'd say `"@type": "URI"` here (and skip the `typeNote`).

Reply:
I had actually forgot we added the URI class. Should we "unpend" it? :)
We could add "uri" code to the normalization we have in place as well.

Reply:
Very reasonable, fixed!